### DISCUSSION TITLE
New on 2.0 TTFB >2s - can this be improved?

### DISCUSSION THREAD

**Discussion by Helmi** [vercel/vercel#4636] [Help] [answered]
Hey,

after finally switching from Now 1.0 to 2.0 I'm trying to improve the response time of our site. While there's tons to improve on the site itself I see a high TTFB (Time to first byte) that is constantly above 2 seconds. Is there any chance to improve that or is that just due to the nature of serverless functions?

Site is based on Nuxtjs just in case that matters.

**Comment by styfle** [comment, marked as answer]
There's a few things you can try:

1. Use SSG to build static files ahead of time and avoid dynamic content if you don't need it. You'll see the greatest performance improvement by switching from SSR to SSG (I believe this is `nuxt generate`).
2. If your serverless functions rely on an upstream database, make sure your functions are running in the same [Origin Region](https://vercel.com/docs/v2/edge-network/regions) by creating a `vercel.json` with the [regions](https://vercel.com/docs/configuration#project/regions) property.
3. Again, any upstream database or other http connections could be the source of slowness, not Vercel. Try a hello world to measure TTFB with something like `module.exports = (req, res) => res.end('hello')`.

If you're still experiencing slowness, [contact support](https://vercel.com/support/request) and we'll take a look at your specific deployment, thanks!

**Reply by Helmi** [reply]
Thanks I'll take it from there.
